### DISCUSSION TITLE
Hi,I like your application very much. But it didn't work well.

### DISCUSSION THREAD

**Discussion by ololee** [kimlimjustin/xplorer#159] [General]
### Description

I am not a `ts` programmer, and I develop `FileExplorer` for my company. I think I still have a better say in this. Your file browser interface is beautiful, but it's terrible to use. Compared with the browser that comes with the operating system, when viewing a large number of files, the speed is very slow, crashes, or even freezes. I think you can try to use C/C++/OC and other higher-performance languages to process the file and return it to ts instead of using ts’ file API. I used java to develop a file browser for Android, and I found that java The File class is too perfect, so that the speed is very poor, so I use C to process the file and return it to java. The speed is improved by about 6 times, but I don't know if it is suitable for ts. Looking forward to your better products.

### Steps To Reproduce

Insert the USB flash drive
Open a catalog of 3000 pictures
Application Not Responsed.....

### Expected behavior

Very smooth display of files, it’s best to use pagination to load

### Xplorer Version

first version

### Operating System Version

Windows 10&&ubuntu 20&&mac mini m1( Big Sur)

### Additional Information

just a user.

**Comment by lg-nineoneninethreenine** [comment]
> I think you can try to use C/C++

Xplorer does use C++ for some Windows modules. Maybe you can try building the app since so many commits occurred since 0.1.0

**Comment by kimlimjustin** [comment]
Yeah, the speed(performance) is undeniably terrible for now, I'm still trying my best to improve it (By calling c/c++ via node API) but can't really find any good resources to learn from. Do you have any good resources to learn file handling like this?

Also, the trash feature is tbh bad approach as it doesn't really integrate the Windows OS (I've googled but can't find any good resources too)

If you got any resources to learn file handling at C/C++, could you please share them? 

Also, if you want to contribute with C/C++, you will be much than welcomed :)

> Xplorer does use C++ for some Windows modules. Maybe you can try building the app since so many commits occurred since 0.1.0

Tbh only some of the function use c++ for now tho, (function like reading directory still use typescript), also the commits is mostly about refactoring code (I think I'll release a newer version at the end of this month).
